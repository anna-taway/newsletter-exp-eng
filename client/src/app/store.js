import { configureStore } from "@reduxjs/toolkit";
import registrationReducer from "../features/userRegistration/userRegistrationSlice";
import productsReducer from '../features/products/productsSlice';


const store = configureStore({
	reducer: {
		registration: registrationReducer,
		products: productsReducer
	},
});

export default store;