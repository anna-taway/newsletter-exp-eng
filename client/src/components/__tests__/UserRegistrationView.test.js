import React from "react";
import UserRegistrationView from "../UserRegistrationView";
import { create } from "react-test-renderer";
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

import { Provider } from 'react-redux';
import store from '../../app/store';

describe("UserRegistrationView component", () => {
  test("UserRegistrationView component matches the snapshot", () => {
    const history = createMemoryHistory()
    const userRegistrationView = create(
        <Provider store={store}>
            <Router history={history}>
                <UserRegistrationView />
            </Router>
        </Provider>
    );
    expect(userRegistrationView.toJSON()).toMatchSnapshot();
  });
});