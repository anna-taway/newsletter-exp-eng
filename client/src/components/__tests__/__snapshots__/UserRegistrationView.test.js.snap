// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRegistrationView component UserRegistrationView component matches the snapshot 1`] = `
<main
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root makeStyles-image-2 MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7"
  />
  <div
    className="MuiPaper-root MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 MuiPaper-elevation6"
  >
    <div
      className="makeStyles-paper-3"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <form
        className="makeStyles-form-5"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
          >
            Username
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="username"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="username"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-10"
              >
                <span>
                  Username
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-12 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                className="PrivateSwitchBase-input-15"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
                value="remember"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Remember me
          </span>
        </label>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-6 MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Sign In
          </span>
        </button>
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Forgot password?
            </a>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Don't have an account? Sign Up
            </a>
          </div>
        </div>
      </form>
      <div
        className="MuiBox-root MuiBox-root-16 makeStyles-box-7"
      >
        <div
          className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
          role="alert"
        >
          <div
            className="MuiAlert-icon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            className="MuiAlert-message"
          >
            Please fill the field
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
